### PR TITLE
Create ens profile setting

### PR DESCRIPTION
**Pull request by pconerly**
This would solve my issue https://github.com/payton/django-siwe-auth/issues/19

I added an option, CREATE_ENS_PROFILE_ON_AUTHN. I tried to emulate CREATE_GROUPS_ON_AUTHN without changing the current behavior for django-siwe-auth users, which is why I default CREATE_ENS_PROFILE_ON_AUTHN to True in backends.py.

I also added some documentation to README.md, but you could cherry-pick that commit and remove a line to include it without the rest of the code here.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -175,6 +175,8 @@ Requirements for using `django-siwe-auth` in a Django application:
    AUTHENTICATION_BACKENDS = ["siwe_auth.backend.SiweBackend"]
    LOGIN_URL = "/"
    SESSION_COOKIE_AGE = 3 * 60 * 60 
+   CREATE_GROUPS_ON_AUTHN = True # defaults to False
```
Comment (payton) [inline README.md:178]: nit: Set to False as per comment

File: README.md [hunk 2]
```diff
@@ -357,9 +359,18 @@ fetch('/api/authentication/nonce', {
 
 
 ### Configuring Your Project
-TODO: Define required and optional configuration variables for project's `settings.py`
-
 
+#### django-siwe-auth specific options in settings.py:
```
Comment (payton) [inline README.md:363]: Please separate these options out into two sections: django-siwe-auth options and django-native options. I want to differentiate between which configurations are custom from django-siwe-auth.

Django Native: AUTH_USER_MODEL, AUTHENTICATION_BACKENDS, LOGIN_URL, SESSION_COOKIE_AGE
Django SIWE Auth: CREATE_GROUPS_ON_AUTHN, CREATE_ENS_PROFILE_ON_AUTHN, CUSTOM_GROUPS, PROVIDER

File: siwe_auth/backend.py [hunk 3]
```diff
@@ -142,14 +143,12 @@ class ENSProfile:
     Container for ENS profile information including but not limited to primary name and avatar.
     """
 
-    name: str
-    avatar: str
+    name: str = ""
```
Comment (payton) [inline siwe_auth/backend.py:146]: Let's make these both default to `None`. The current behavior is that `ENS.fromWeb3(w3).name(address=ethereum_address)` will return `None` if a given request is not found.

**Review by payton — CHANGES_REQUESTED** [review]
Looks great @pconerly. Just a left a few minor changes to be made.

**Comment by pconerly** [comment]
The changes you asked for make sense, thanks for reviewing it. I just pushed a commit with those changes

**Review by payton — APPROVED** [review]
Awesome. Thanks @pconerly !

**Pull request merged**